ack/readers/awin_advertiser/reader.py

```diff
@@ -40,20 +40,32 @@ def __init__(
     @retry
     def request(self):
         if self.report_type == REPORT_TYPES[0]:
-            build_url = f"{AWIN_API_ENDPOINT}{self.advertiser_id}"\
-                f"/reports/{self.report_type}?startDate={self.start_date}&endDate={self.end_date}"\
-                f"&timezone={self.timezone}&accessToken={self.auth_token}"
+            payload = {
+                'startDate': self.start_date, 
+                'endDate': self.end_date, 
+                'timezone': self.timezone, 
+                'accessToken': self.auth_token
+                }
+            build_url = f"{AWIN_API_ENDPOINT}{self.advertiser_id}/reports/{self.report_type}?"
```

Comment:
As build_url is common to the three cases so you can make only one declaration outside the if statement, your code will be cleaner ;)